File: frontend/src/Types/meeting.ts
import type { Guid } from './api';

export type MeetingType = 0 | 1 | 2 | 3 | 4;
export type MeetingStatus = 0 | 1 | 2;

export interface Meeting {
  id: Guid;
  title: string;
  date: string;
  type: MeetingType;
  status: MeetingStatus;
  sprintId: Guid | null;
  notes: string | null;
  decisions: string | null;
  actions: string | null;
  participantIds: Guid[];
  dailyMeetingEntryIds: Guid[];
}

export interface CreateMeetingRequest {
  title: string;
  date: string;
  type: MeetingType;
  sprintId?: Guid | null;
  notes?: string | null;
  decisions?: string | null;
  actions?: string | null;
}

export interface UpdateMeetingRequest {
  title: string;
  date: string;
  type: MeetingType;
  status: MeetingStatus;
  sprintId?: Guid | null;
  notes?: string | null;
  decisions?: string | null;
  actions?: string | null;
}

File: backend/src/ScrumLog.Api/Data/Entities/Meeting.cs
namespace ScrumLog.Api.Data.Entities;

public class Meeting
{
    public Guid Id { get; set; }

    public required string Title { get; set; }

    public DateTime Date { get; set; }

    public MeetingType Type { get; set; }

    public Guid SprintId { get; set; }

    public Sprint Sprint { get; set; } = null!;

    public string? Notes { get; set; }

    public string? Decisions { get; set; }

    public string? Actions { get; set; }

    public ICollection<MeetingParticipant> Participants { get; set; } = new List<MeetingParticipant>();

    public ICollection<DailyMeetingEntry> DailyMeetingEntries { get; set; } = new List<DailyMeetingEntry>();
}
